src/Nethermind/Nethermind.Blockchain/Bloom/FixedSizeFileStore.cs

```diff
@@ -0,0 +1,88 @@
+﻿//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.IO;
+
+namespace Nethermind.Blockchain.Bloom
+{
+    public class FixedSizeFileStore : IFileStore
+    {
+        private readonly string _path;
+        private readonly int _elementSize;
+        private readonly FileStream _fileWrite;
+        private readonly FileStream _fileRead;
+        
+        public FixedSizeFileStore(string path, int elementSize)
+        {
+            _path = path;
+            _elementSize = elementSize;
+            _fileWrite = File.Open(path, FileMode.OpenOrCreate, FileAccess.Write, FileShare.Read);
+            _fileRead = File.Open(path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+        }
+
+        public void Write(long index, byte[] element)
+        {
+            if (element.Length != _elementSize)
+            {
+                throw new ArgumentException($"Element size incorrect. Only elements of length {_elementSize} are acceptable.");
+            }
+            
+            lock (_fileWrite)
+            {
+                SeekIndex(_fileWrite, index);
+                _fileWrite.Write(element, 0, _elementSize);
```

Comment:
Write(element.AsSpan())

Reply:
public override void Write(byte[] array, int offset, int count)
        {
            ValidateReadWriteArgs(array, offset, count);
            if (_useAsyncIO)
            {
                WriteAsyncInternal(new ReadOnlyMemory<byte>(array, offset, count), CancellationToken.None).GetAwaiter().GetResult();
            }
            else
            {
                WriteSpan(new ReadOnlySpan<byte>(array, offset, count));
            }
        }

Reply:
public override void Write(ReadOnlySpan<byte> buffer)
        {
            if (GetType() == typeof(FileStream) && !_useAsyncIO)
            {
                if (_fileHandle.IsClosed)
                {
                    throw Error.GetFileNotOpen();
                }
                WriteSpan(buffer);
            }
            else
            {
                // This type is derived from FileStream and/or the stream is in async mode.  If this is a 
                // derived type, it may have overridden Write(byte[], int, int) prior to this Write(ReadOnlySpan<byte>)
                // overload being introduced.  In that case, this Write(ReadOnlySpan<byte>) overload should use the behavior
                // of Write(byte[],int,int) overload.  Or if the stream is in async mode, we can't call the
                // synchronous WriteSpan, so we similarly call the base Write, which will turn delegate to
                // Write(byte[],int,int), which will do the right thing if we're in async mode.
                base.Write(buffer);
            }
        }